### DISCUSSION TITLE
Occurrences of a recurring event

### DISCUSSION THREAD

**Discussion by dmitrysync** [ecederstrand/exchangelib#970] [General]
How can I get all occurrences by RecurringMaster?

**Comment by ecederstrand** [comment]
There are a couple of different options, all described in https://ecederstrand.github.io/exchangelib/#recurring-calendar-items

**Reply by dmitrysync** [reply]
Thanks. I saw this chapter.
But it doesn't help me.

Can I pass the id of the recurring event to `view()`?

My goal is to get all occurrences of a recurring event in a time span.
The `view()` method copes with this in half, since you cannot specify the identifier of a recurring event.

**Reply by ecederstrand** [reply]
EWS doesn't support getting items by ID in the `view()` method. There are two ways to overcome this:

1. Filter for the timespan that your master occurrence is, and pick out the occurrences that belong to the master recurrence
2. Find out how many recurrences you have in total, and use `CalendarItem.occurrence()` to fetch each of them.

**Reply by dmitrysync** [reply]
About the second way. How can I find out the number of occurrences?

**Reply by ecederstrand** [reply]
You can look at the `CalendarItem.recurrence` object.

**Reply by ecederstrand** [reply]
You probably want to take `CalendarItem.deleted_occurrences` into account. Some trial-and-error will tell you how they work together.

**Reply by dmitrysync** [reply]
Thanks a lot.
